File: Employee_Management_System-main/frontend/src/Services/DepartmentService.js
import api from "./api";

const API_URL = "/departments";

// GET all departments
export const getDepartments = () => {
    return api.get(API_URL);
};

// GET department by ID
export const getDepartmentById = (id) => {
    return api.get(`${API_URL}/${id}`);
};

// POST — add new department
export const addDepartment = (department) => {
    return api.post(API_URL, department);
};

// PUT — update existing department
export const updateDepartment = (id, department) => {
    return api.put(`${API_URL}/${id}`, department);
};

// DELETE — remove department
export const deleteDepartment = (id) => {
    return api.delete(`${API_URL}/${id}`);
};

// GET total department count
export const getDepartmentCount = () => {
    return api.get(`${API_URL}/count`);
};

File: Employee_Management_System-main/backend/src/main/java/com/ems/service/DepartmentService.java
package com.ems.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.ems.entity.Department;
import com.ems.repository.DepartmentRepository;

@Service
public class DepartmentService {

    @Autowired
    private DepartmentRepository repository;

    // Get all departments
    public List<Department> getAllDepartments() {
        return repository.findAll();
    }

    // Save a new department
    public Department saveDepartment(Department department) {
        return repository.save(department);
    }

    // Get department by ID
    public Department getDepartmentById(Integer id) {
        return repository.findById(id).orElse(null);
    }

    // Update department
    public Department updateDepartment(Integer id, Department department) {

        Department existing = repository.findById(id).orElse(null);

        if (existing != null) {
            existing.setDepartmentName(department.getDepartmentName());
            existing.setDescription(department.getDescription());
            existing.setLocation(department.getLocation());
            return repository.save(existing);
        }

        return null;
    }

    // Delete department
    public void deleteDepartment(Integer id) {
        repository.deleteById(id);
    }

    // Count total departments
    public long countDepartments() {
        return repository.count();
    }
}
